File: src/Entity/DonationM.php
<?php

namespace App\Entity;

use App\Repository\DonationMRepository;
use Doctrine\DBAL\Types\Types;
use Doctrine\ORM\Mapping as ORM;
use Symfony\Component\Validator\Constraints as Assert;


#[ORM\Entity(repositoryClass: DonationMRepository::class)]
class DonationM
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column(name:'donationMId')]
    private ?int $donationMId = null;

    #[ORM\Column]
    #[Assert\GreaterThanOrEqual(value: 10, message: "The donation amount should be equal to or greater than 10 DT.")]
    #[Assert\NotBlank( message: "You must enter the donation amount")]


    private ?float $Donation_Amount = null;

    #[ORM\Column(type: Types::DATE_MUTABLE)]
    #[Assert\EqualTo("today", message: "You must make a donation today")]
    #[Assert\NotBlank( message: "You must enter the donation date")]
    private ?\DateTimeInterface $donationM_Date = null;

    #[ORM\ManyToOne(inversedBy: 'donationMs')]
    #[ORM\JoinColumn(nullable: false, name: 'Account_Key', referencedColumnName: 'accountId')]
    #[Assert\NotNull(message: "You must select an account")]
    private ?Account $Account_Key = null;

    public function getdonationMId(): ?int
    {
        return $this->donationMId;
    }

    public function getDonationAmount(): ?float
    {
        return $this->Donation_Amount;
    }

    public function setDonationAmount(float $Donation_Amount): static
    {
        $this->Donation_Amount = $Donation_Amount;

        return $this;
    }

    public function getDonationMDate(): ?\DateTimeInterface
    {
        return $this->donationM_Date;
    }

    public function setDonationMDate(\DateTimeInterface $donationM_Date): static
    {
        $this->donationM_Date = $donationM_Date;

        return $this;
    }

    public function getAccountKey(): ?Account
    {
        return $this->Account_Key;
    }

    public function setAccountKey(?Account $Account_Key): static
    {
        $this->Account_Key = $Account_Key;

        return $this;
    }
}

File: src/main/java/entities/DonationM.java
package entities;

import java.util.Date;

public class DonationM {
    private int donationMId;
    private float donationAmount;
    private Date donationMDate;
    private int accountKey;

    public DonationM(int donationMId, float donationAmount, Date donationMDate, int accountKey) {
        this.donationMId = donationMId;
        this.donationAmount = donationAmount;
        this.donationMDate = donationMDate;
        this.accountKey = accountKey;
    }

    public DonationM(float donationAmount, Date donationMDate, int accountKey)
    {
        this.donationAmount=donationAmount;
        this.donationMDate=donationMDate;
        this.accountKey=accountKey;
    }

    public DonationM() {
    }

    public int getDonationMId() {
        return donationMId;
    }

    public void setDonationMId(int donationMId) {
        this.donationMId = donationMId;
    }

    public float getDonationAmount() {
        return donationAmount;
    }

    public void setDonationAmount(float donationAmount) {
        this.donationAmount = donationAmount;
    }

    public java.sql.Date getDonationMDate() {
        return new java.sql.Date(donationMDate.getTime());
    }

    public void setDonationMDate(Date donationMDate) {
        this.donationMDate = donationMDate;
    }

    public int getAccountKey() {
        return accountKey;
    }

    public void setAccountKey(int accountKey) {
        this.accountKey = accountKey;
    }

    @Override
    public String toString() {
        return "donationM{" +
                "donationMId=" + donationMId +
                ", donationAmount=" + donationAmount +
                ", donationMDate='" + donationMDate + '\'' +
                ", accountKey=" + accountKey +
                '}';
    }
}
